### PR TITLE
Change the path separator variables' type to public

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2831
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec

### CODE REVIEW

**Comment by niveathika** [comment]
@kalaiyarasiganeshalingam Lets update the changelog too, 

"Expose path separator constants outside the package."

IMO better if we document this in the Module md too

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 org = "ballerina"
 name = "file"
-version = "1.2.1"
+version = "1.2.2"
```
Comment (daneshk) [inline ballerina/Ballerina.toml:4]: Shall we change this to the next minor release?. We need to do minor release on top of update1

Comment (niveathika) [reply ballerina/Ballerina.toml:4]: If all the breaking changes are reverted, should we update to update 1 dependency?

Comment (daneshk) [reply ballerina/Ballerina.toml]: If it is required only we need to release it on top of update1. any this is an API addition. better we go with minor release

**Comment by daneshk** [comment]
> @kalaiyarasiganeshalingam Lets update the changelog too,
> 
> "Expose path separator constants outside the package."
> 
> IMO better if we document this in the Module md too

Yes, we need to update the documentation and spec as well. @kalaiyarasiganeshalingam can you check. This change should come as a proposal

**Comment by kalaiyarasiganeshalingam** [comment]
> > @kalaiyarasiganeshalingam Lets update the changelog too,
> > "Expose path separator constants outside the package."
> > IMO better if we document this in the Module md too
> 
> Yes, we need to update the documentation and spec as well. @kalaiyarasiganeshalingam can you check. This change should come as a proposal

Please find the proposal for this: https://github.com/ballerina-platform/ballerina-standard-library/issues/2834

**Comment by daneshk** [comment]
> > > @kalaiyarasiganeshalingam Lets update the changelog too,
> > > "Expose path separator constants outside the package."
> > > IMO better if we document this in the Module md too
> > 
> > 
> > Yes, we need to update the documentation and spec as well. @kalaiyarasiganeshalingam can you check. This change should come as a proposal
> 
> @daneshk @niveathika Please find the proposal for this: [ballerina-platform/ballerina-standard-library#2834](https://github.com/ballerina-platform/ballerina-standard-library/issues/2834)

The proposal looks good to me, shall we update the spec and documentation in this PR

**Comment by kalaiyarasiganeshalingam** [comment]
> > > > @kalaiyarasiganeshalingam Lets update the changelog too,
> > > > "Expose path separator constants outside the package."
> > > > IMO better if we document this in the Module md too
> > > 
> > > 
> > > Yes, we need to update the documentation and spec as well. @kalaiyarasiganeshalingam can you check. This change should come as a proposal
> > 
> > 
> > @daneshk @niveathika Please find the proposal for this: [ballerina-platform/ballerina-standard-library#2834](https://github.com/ballerina-platform/ballerina-standard-library/issues/2834)
> 
> The proposal looks good to me, shall we update the spec and documentation in this PR

Updated.

File: docs/spec/spec.md [hunk 2]
```diff
@@ -44,7 +44,9 @@ The conforming implementation of the specification is released and included in t
 
 # 1. Overview
 Ballerina file standard library provides functionalities related to manipulating and working with files and directories.
-All operations are supported on both Windows and Unix-based operating systems. 
+All operations are supported on both Windows and Unix-based operating systems. It provides the following separators as constants which are widely used in file path creation:
+-  `file:pathSeparator`: It is a character used to separate the parent directories that make up the path to a specific location. For windows, it’s ‘\’ and for UNIX it’s ‘/’
+-  `file:pathListSeparator`: It is a character commonly used by the operating system to separate paths in the path list. For windows, it’s ‘;‘ and for UNIX it’s ‘:’
```
Comment (daneshk) [inline docs/spec/spec.md:49]: Shall we put this under path operations?

```
## 4.1 Path Constants
OS-specific path constants

-  `pathSeparator`: It is a character used to separate the parent directories that make up the path to a specific location. For windows, it’s ‘\’ and for UNIX it’s ‘/’
-  `pathListSeparator`: It is a character commonly used by the operating system to separate paths in the path list. For windows, it’s ‘;‘ and for UNIX it’s ‘:’
```

File: ballerina/Module.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 This module provides APIs to create, delete, rename the file/directory, retrieve metadata of the given file, and manipulate the file paths in a way that is compatible with the operating system, and a `Directory Listener`, which is used to listen to the file changes in a directory in the local file system.
 
+This also provides the following separators which are widely used in file path creation:
+-  `file:pathSeparator`: It is a character used to separate the parent directories that make up the path to a specific location. For windows, it’s ‘\’ and for UNIX it’s ‘/’
+-  `file:pathListSeparator`: It is a character commonly used by the operating system to separate paths in the path list. For windows, it’s ‘;‘ and for UNIX it’s ‘:’
+
```
Comment (daneshk) [inline ballerina/Module.md:8]:
```suggestion
```

Comment (daneshk) [reply ballerina/Module.md:8]: just thinking about whether we need to mention it here

**Review by daneshk — APPROVED** [review]

**Pull request merged**